### PR TITLE
services/horizon/internal/actions: Use uint32 for ledger_id and transform error.

### PR DESCRIPTION
**Pull request by abuiles**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Extend `GetParams` to handle `schema` errors. 

### Why

`govalidator` is great for string query parameters, however, if the value represents something different to a string you have to add custom code to cast the value to their intended type, for example `string` to `uint`. 

This PR extends how we map query params to the destination struct so if there is a problem while using the query params to fill the params struct, we report the failing field and show the error, allowing us to add custom errors if necessary.

### CODE REVIEW

File: services/horizon/internal/actions/validators.go [hunk 1]
```diff
@@ -31,7 +29,7 @@ var customTagsErrorMessages = map[string]string{
 	"amount":          "Amount must be positive",
 	"asset":           "Asset must be the string \"native\" or a string of the form \"Code:IssuerAccountID\" for issued assets.",
 	"assetType":       "Asset type must be native, credit_alphanum4 or credit_alphanum12",
-	"ledgerID":        "Ledger ID must be higher than 0",
+	"ledger_id":       "Ledger ID must be higher than 0",
```
Comment (bartekn) [inline services/horizon/internal/actions/validators.go:32]: Let's keep it consistent. snake_case or camelCase. Personally, I think snake_case is better here but I'm OK with any.

Comment (abuiles) [reply services/horizon/internal/actions/validators.go:32]: @bartekn the reason why it is `snake_case` here is that it needs to match exactly to the query param name --  for this particular scenario we are not using govalidator but rather the schema failing to parse the param.

**Comment by abuiles** [comment]
@bartekn yeah, we can use `bool` - my only concern is that using bool will take on more values than `true` or `false` - I'm okay doing that but then this query params can take on other values that represent truth like `1`

**Comment by abuiles** [comment]
I'll do the same for booleans and register a better error message for types. So we can have a custom message if the field matches they query parameter or if the field is of a given type.

**Comment by abuiles** [comment]
I added support for `bool` too.

**Pull request merged**